Organize the endpoints.py file

Guptacos opened the pull request:
- Update playlist URIs to use `%s` formatting instead of `{}`
- Organize endpoints.py

Guptacos committed:
Fix minor style problems to pass linter

- Fix style in spotifython, tests, and examples dirs
- Change the method of getting filenames for the linter. Now uses
  a python script instead of bash commands. This is because Makefile
  $(shell) commands don't play well with '$' in a regex string.

Guptacos committed:
Remove response.py

We no longer need a Response object.

Guptacos committed:
Add refresh() and raw() to User

Also fix the _update_fields method to also try to update private fields.

Guptacos committed:
Skip most tests, as most are not yet implemented

Guptacos committed:
Change most instances of list(map())

[func(elem) for elem in elems] is more readable than list(map(func, elems))

Guptacos committed:
Change method naming in player_control.py

get_current_user -> current_user
image['url'] -> image.url()

Guptacos committed:
Standardize init's docstring in all classes

Guptacos committed:
Standardize all first line docstrings

- Single line dostrings have start and end quotes in the same line
- First line of docstrings have period at the end
- Standardize some other formatting, like whitespace and method argument
  positions

Guptacos committed:
Format library to look better with Sphinx

- Standardize all first line docstrings
    - Single line dostrings have start and end quotes in the same line
    - First line of docstrings have period at the end
    - Standardize some other formatting, like whitespace and method
      argument positions
- Fix Sphinx compilation warnings
    - Add whitespace between single line sub-bullets in docstrings so
      that it formats correctly in output html.
- Remove __init__ from documentation
    - Instead of showing a constructor that says "do not use", describe
      at top level how to get an instance.
    - Hyperlink method names at the top level of each class's
      documentation.
    - conf.py: shorten display names for methods and classes. Now omits
      package and filename when displaying a class / method in the html.
- Format docstrings to fix how they look when rendered by Sphinx
- Fix rst links to use __ instead of _, making them "anonymous links" so
  that Sphinx doesn't complain.
- Add extra section to session.rst. Now displays SearchResult as its own
  class on the same page as Session.

Guptacos committed:
Remove BASE_URI duplicates in Playlist class

utils.request() already prepends BASE_URI to the input endpoint, so
Playlist shouldn't also do that.

Guptacos committed:
Organize the endpoints.py file

Guptacos committed:
Organize the endpoints.py file

Guptacos committed:
Add dunder methods to autogenerated docs

Guptacos committed:
Break up classes into separate files

Guptacos committed:
Add terms page

The terms page acts as a sorted glossary that can be referenced in other
documentation. To reference a term (for this example, we'll use market):

:term:`any text you want in the link <Market>`

Note that the term name must mach (case sensitive) the term you are
trying to link to.

Guptacos committed:
Change references to market or track relinking

All references now point to the terms page, centralizing repeated
documentation. I may have missed some, but I got most of them.

spenceryu reviewed (dismissed):
You should probably update the PR to note that playlist URIs are also fixed here.

rmannava commented on spotifython/endpoints.py:
can we remove 'GET' from all the endpoints? not every endpoint has it when it should like top tracks, and it would be cleaner to just remove every get

rmannava commented on spotifython/endpoints.py:
user and playlist are repeated endpoints

Guptacos committed:
Change how Github actions tests our code

Using `python3 -m unittest discover -s test` recursively finds tests and
runs extra things we don't want it to run. Just use `python3 -m
unittest` instead.

Commit:
add playlist track relinking support (#95)

spenceryu committed:
Tests for Session class (#99)

spenceryu committed:
Testing with mocks for artist (#98)

* get basic equality tests to work, add logging ability

* remove logging using logger, it is already supported

* add _update_fields() test

* add test for top_tracks

* add test for related_artists

* add artist_test

* fix linter errors

* add proper paging limits to paginate search, fix tests and validation for artist.albums()

* modify paginate_get to enable unlimited items to be returned

* fix artist.albums call with updated pagination

spenceryu committed:
docs for track relinking search limitations (#100)

Commit:
create playlist tests outline (#96)

* create playlist tests outline

* Remove response.py

We no longer need a Response object.

* Change method naming in player_control.py

get_current_user -> current_user
image['url'] -> image.url()

* Remove BASE_URI duplicates in Playlist class

utils.request() already prepends BASE_URI to the input endpoint, so
Playlist shouldn't also do that.

* Change most instances of list(map())

[func(elem) for elem in elems] is more readable than list(map(func, elems))

* Add refresh() and raw() to User

Also fix the _update_fields method to also try to update private fields.

* Standardize init's docstring in all classes

* Standardize all first line docstrings

- Single line dostrings have start and end quotes in the same line
- First line of docstrings have period at the end
- Standardize some other formatting, like whitespace and method argument
  positions

* Skip most tests, as most are not yet implemented

* Add dunder methods to autogenerated docs

* Break up classes into separate files

* Fix minor style problems to pass linter

- Fix style in spotifython, tests, and examples dirs
- Change the method of getting filenames for the linter. Now uses
  a python script instead of bash commands. This is because Makefile
  $(shell) commands don't play well with '$' in a regex string.

* Format library to look better with Sphinx

- Standardize all first line docstrings
    - Single line dostrings have start and end quotes in the same line
    - First line of docstrings have period at the end
    - Standardize some other formatting, like whitespace and method
      argument positions
- Fix Sphinx compilation warnings
    - Add whitespace between single line sub-bullets in docstrings so
      that it formats correctly in output html.
- Remove __init__ from documentation
    - Instead of showing a constructor that says "do not use", describe
      at top level how to get an instance.
    - Hyperlink method names at the top level of each class's
      documentation.
    - conf.py: shorten display names for methods and classes. Now omits
      package and filename when displaying a class / method in the html.
- Format docstrings to fix how they look when rendered by Sphinx
- Fix rst links to use __ instead of _, making them "anonymous links" so
  that Sphinx doesn't complain.
- Add extra section to session.rst. Now displays SearchResult as its own
  class on the same page as Session.

* Change how Github actions tests our code

Using `python3 -m unittest discover -s test` recursively finds tests and
runs extra things we don't want it to run. Just use `python3 -m
unittest` instead.

* Add terms page

The terms page acts as a sorted glossary that can be referenced in other
documentation. To reference a term (for this example, we'll use market):

:term:`any text you want in the link <Market>`

Note that the term name must mach (case sensitive) the term you are
trying to link to.

* Change references to market or track relinking

All references now point to the terms page, centralizing repeated
documentation. I may have missed some, but I got most of them.

* add playlist track relinking support (#95)

* fix playlist track parsing

* create playlist tests outline

* fix playlist track parsing

Co-authored-by: Niko Gupta <niko.gupta@gmail.com>

spenceryu committed:
Merge branch 'organize-endpoints' of https://github.com/Guptacos/spotifython into organize-endpoints

spenceryu committed:
remove GET from all endpoints

spenceryu commented on spotifython/endpoints.py:
Not coupling these two methods is easier for us to maintain in the interest of future maintainability.
